File: tools/bench/answers/go_haiku/s2/12_celsius.go
package main

import (
	"fmt"
)

func main() {
	temps := []float64{12.5, 30.0, -5.0}
	maxF := 0.0
	for i, c := range temps {
		f := c*9/5 + 32
		if i == 0 || f > maxF {
			maxF = f
		}
	}
	fmt.Println(maxF)
}

File: tools/bench/answers/rust_haiku/s2/12_celsius.rs
fn main() {
    let temps = [12.5, 30.0, -5.0];

    let mut max_f = f64::NEG_INFINITY;

    for c in temps {
        let f = c * 9.0 / 5.0 + 32.0;
        if f > max_f {
            max_f = f;
        }
    }

    println!("{}", max_f);
}
